lib/teleterm/apiserver/handler/handler_device_trust.go

```diff
@@ -25,15 +25,19 @@ import (
 )
 
 func (s *Handler) AuthenticateWebDevice(ctx context.Context, req *api.AuthenticateWebDeviceRequest) (*api.AuthenticateWebDeviceResponse, error) {
-	clusterURI, err := uri.Parse(req.GetRootClusterUri())
-	if err != nil {
-		return nil, trace.Wrap(err)
+	switch {
+	case req.GetRootClusterUri() == "":
+		return nil, trace.BadParameter("root_cluster_uri required")
+	case req.DeviceWebToken.GetId() == "":
+		return nil, trace.BadParameter("device_web_token.id required")
+	case req.DeviceWebToken.GetToken() == "":
+		return nil, trace.BadParameter("device_web_token.token required")
 	}
 
-	daemonResponse, err := s.DaemonService.AuthenticateWebDevice(ctx, clusterURI, req)
+	clusterURI, err := uri.Parse(req.GetRootClusterUri())
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
-	return &api.AuthenticateWebDeviceResponse{ConfirmationToken: daemonResponse.ConfirmationToken}, nil
+	return s.DaemonService.AuthenticateWebDevice(ctx, clusterURI, req)
```

Comment:
If you want to keep trace errors:

```suggestion
	resp, err := s.DaemonService.AuthenticateWebDevice(ctx, clusterURI, req)
	return resp, trace.Wrap(err)
```